### ISSUE TITLE
WebGui?

### ISSUE THREAD

**Issue by Budman17r** [arkmanager/ark-server-tools#255]
Just curious, I appreciate the response, So far loving this thing.

**Comment by klightspeed** [comment]
This has been suggested a few times, and I thought I remembered that someone had created a basic web UI, but I can't find it.

@FezVrasta suggested it in #57, and @Sispheor suggested it in #64.

**Comment by Budman17r** [comment]
I appreciate the feedback I should have searched harder.

**Comment by klightspeed** [comment]
Keep this around as a feature request, since there is no separate issue for it.

**Comment by verticalavenue** [comment]
We use webmin and usermin for the web interface for the server. You can set up all the commands in webmin and make them accessible via steam user on usermin. I run multiple servers so the custom commands run all the games rather than one interface with each game.

**Comment by benzon** [comment]
Did a hacky web interface that currently only works on servers with Sudoers since we gotta allow www-data to run the arkmanager command, and add the www-data to the steam group and do the correct chmod for it to work, but it works :) even shows live update of what is happening, and do a check to se if the server is online, before saying it's online but again was just created for my community, but it's quite easy :)

**Comment by benzon** [comment]
Currently cleaning up my WebUI for Ark it's based on JavaScript, php, reason for php is thats the language i know the best, and you can use easyengine to install a Nginx webserver quite easy.

Currently learning NodeJS so in the future it might be NodeJS based, to remove permission issues.

**Comment by FezVrasta** [comment]
I would integrate a web UI only if written in Node.js or any other programming language that doesn't need to install heavy dependencies on the server.

@benzon if you need any help with a Node.js conversion feel free to ask me in this issue, I've _a bit_ of experience with it.

**Comment by benzon** [comment]
@FezVrasta Agree, the version im doing now is to polish what i got and clean up the code so i can allow others to actually download and use it, the issue with it is dependencie on Sudoers, and Nginx/EasyEngine(Greate Nginx Tool) and some permission work like adding www-data to the steam group.

NodeJS would remove the permission issues we would be able to run it as the Steam user without any major security issues.

About the API thought not sure it's that crucial for the WebUI should still be simple to create with out since NodeJS should be able to run CMD commands and grap the output live from arkmanager, and that data should be fine, since it tells whats going on.

Still learning NodeJS, but my goal is to convert the HTML, JavaScript, PHP WebUI in to NodeJS so it's easy to deploy and avoid permission issues on Linux.

**Comment by benzon** [comment]
The only thing php currently do is run the CMD commands live load the reply and wright it to a text file.

And then it uses Source Query PHP  class to grap the status of the game server if its up or down, and show a player list on the page, it's a single page design where the text file and calls to php is done with Ajax.

**Comment by Lihis** [comment]
Do you guys think there is need for android app? As I'm might be interested to contribute this too. I have wrote little API for myself with php too and made Android app which uses that API. I made those because I wanted to monitor server "on-the-road" easier way rather than writing commands via ConnectBot app. With my API/app I'm able to check server status (is it running, is it online and version), chat log (last 500 messages), player names (and current steam name as ARK reports only the steam name which was when the user first time joined server). I planned to add features like sending messages/broadcasts, kicking and banning players from app too.

I chose php because the server is running web server anyway so the huge dependencies was not problem but I agree that the nodeJS would be a better way to go.

@FezVrasta drop me email if you're interested to test the app, I can send it to you so you can test it with my server.

**Comment by FezVrasta** [comment]
(@benzon, when I was talking about writing some official API :P)

@Lihis I think it's a great idea, but I would love to get an official API supported by the contributors before chime into specific apps.
I can prepare a basic API interface in Node.js, easily extensible, and then all the interested guys can contribute to it.

Once a Node.js API is ready, everybody can easily write 3rd party apps and interact with it.

Do we have a plan so?

**Comment by Lihis** [comment]
@FezVrasta Yeah, API first (well documented? :p) then it is easy to start develop 3rd party apps, like my app or this webgui thing.

I do not have experience with nodeJS but I took a quick look and it does not seem to be too hard to learn for me. So count me in.

**Comment by FezVrasta** [comment]
I've pushed to the new `api` branch a simple REST server.

To run it `cd` in `/api` and then

````
npm install
npm start
```

https://github.com/FezVrasta/ark-server-tools/tree/api

**Comment by FezVrasta** [comment]
@klightspeed if we want to support this kind of stuff we need to get JSON from Arkmanager, is it something easy to achieve?

Thanks!

**Comment by Lihis** [comment]
That was fast. I got that up and running. By the way, are you running Ubuntu?

**Comment by FezVrasta** [comment]
No I'm on OS X.

**Comment by Lihis** [comment]
Whoops.. Missed the "on your server" part from my message..

Anyways, this is irrelevant at this point but on the package.json line 7 ``node`` needs to be changed to ``nodejs`` at least for Debian 8.2 and Ubuntu 14.04 (and upwards). As the "nodejs" package provides ``/usr/bin/nodejs``. Or the user must install ``nodejs-legacy`` package which provides ``/usr/bin/node`` or just create symlink. On Arch Linux "nodejs" package seems to be providing "node" binary.

**Comment by FezVrasta** [comment]
Ah, this sucks. I wasn't aware of this problem. On Windows and OS X the binary is just `node`.

May you trying adding a `start.sh` to the `api` folder and put inside it these lines?
```
if [ "$(type -t node)" = "file" ]; then
    node index.js
else
    nodejs index.js
fi
}```

and put `bash start.sh` as `start` command?

It should use the correct binary

**Comment by FezVrasta** [comment]
Please for anything related to the API use the new #335 issue.

**Comment by FezVrasta** [comment]
Ok I've pushed the changes.
